src/internal/collection/postgres_collection.go

```diff
@@ -214,6 +214,31 @@ func (c *postgresReadWriteCollection) GetByIndex(index *Index, indexVal string,
 	return c.getByIndex(context.Background(), c.tx, index, indexVal, val, opts, true, f)
 }
 
+func (c *postgresCollection) getUniqueByIndex(ctx context.Context, q sqlx.ExtContext, index *Index, indexVal string, val proto.Message) error {
```

Comment:
Added a convenience function to collections to get a unique row by a secondary index.  If this finds more than one row, an error is returned.